### PR TITLE
Add ReportException(Exception e, Dictionary<> props) for BL-2418

### PR DESCRIPTION
**Pull request by StephenMcConnel**
More information needs to be passed with Exception Analytics.
Also fixed for compiling on Linux.

### CODE REVIEW

File: src/DesktopAnalytics/Analytics.cs [hunk 1]
```diff
@@ -230,14 +230,31 @@ public static void Track(string eventName, Dictionary<string, string> properties
 		/// <param name="e"></param>
 		public static void ReportException(Exception e)
 		{
+			ReportException(e, null);
```
Comment (JohnThomson) [inline src/DesktopAnalytics/Analytics.cs]: Is this project using a late enough version of C# so we could just make the new parameter optional?

**Comment by JohnThomson** [comment]
LGTM with one minor suggestion

File: src/DesktopAnalytics/Analytics.cs (hunk 1 above)
Comment (StephenMcConnel) [inline src/DesktopAnalytics/Analytics.cs]: I thought about using an optional parameter, but doing so breaks ABI compatibility, which may not be an important consideration in this circumstance.  (and I mean the B for Binary in ABI, although i'm not sure that's exactly the right term for the concept I was reading about in the discussion of optional parameters.)

**Comment by hatton** [comment]
If it's ok with y'all, I'm going to merge this and then switch to the optional paramer

**Pull request merged**